iFixit/react-commerce
# TOC: Polish 2

QA
---

* Update feedback on https://github.com/iFixit/ifixit/discussions/48967#discussioncomment-6870061
* Ensure the Prod version looks around the same

>1. First steps only seems to be highlighted in the TOC when the header First Steps is visible. It makes for a brief flash of highlighting as you scroll down. It looks like the other sections are highlighted when their headers are out of view.

Mostly solved. This was due to the causes list appearing when the TOC was enabled on desktop. The causes list will now only appear on mobile

>2. Related problems goes nowhere and is never highlighted. I assume this is WIP, but I'm not excluding anything here.

Solved.

>3. Scrolling issue in comment above is present on desktop as well.

Not solved. This is due to the browser controlling scroll vs my code controlling scroll. If you click a header, the browser determines which pixel needs to be shown. If you click the TOC, I am determining which pixel to scroll into view.

>4. Mobile First steps #Section_First_Steps doesn't accommodate the height of the header, so it's obscured by the TOC

I am not sure this is something we can solve for. Namely, the math to scroll to an item does not take into account the height of the header. In fact, it does not know about a header existing.

>5. I don't love it when cut off text can be so close to other cut off text.

I added back a fade on mobile, but we can remove it if Mattia is not a fan. Personally, I like it as well.

>6. I can see a pixel or two of scrolling above the TOC.

I think this is solved now that we no longer have the scroll percentage at the top of the page
